applications/CoSimulationApplication/tests/test_cosim_EMPIRE_API.py

```diff
@@ -0,0 +1,290 @@
+
+def GetFilePath(fileName):
+    return os.path.join(os.path.dirname(os.path.realpath(__file__)), fileName)
+
+class TestCoSim_EMPIRE_API(KratosUnittest.TestCase):
+
+    def test_unused_fcts(self):
+        KratosCoSim.EMPIRE_API.EMPIRE_API_Connect("dummy.xml")
+        KratosCoSim.EMPIRE_API.EMPIRE_API_Disconnect()
+        KratosCoSim.EMPIRE_API.EMPIRE_API_getUserDefinedText("dummy_element")
+
+    def test_EMPIRE_API_sendConvergenceSignal(self):
+        with self.assertRaisesRegex(RuntimeError, "Input can only be 0 for non-convergence or 1 for convergence, called with: 2"):
+            KratosCoSim.EMPIRE_API.EMPIRE_API_sendConvergenceSignal(2)
+
+        signal = 1 # means convergence
+
+        KratosCoSim.EMPIRE_API.EMPIRE_API_sendConvergenceSignal(signal)
+
+        self.__CheckConvergenceSignalFile(signal)
+
+        os.remove(conv_signal_file_name)
+
+    def test_EMPIRE_API_recvConvergenceSignal(self):
+        # write file with convergence info manually
+        signal = 0 # means non-convergence
+
+        with open(conv_signal_file_name, 'w') as conv_signal_file:
+            conv_signal_file.write(str(signal))
+
+        self.__CheckConvergenceSignalFile(signal) # to make sure that the file was written correctly
+
+        is_converged = KratosCoSim.EMPIRE_API.EMPIRE_API_recvConvergenceSignal()
+
+        self.assertEqual(signal, is_converged)
+        # make sure that the file was deleted
+        self.assertFalse(os.path.isfile(conv_signal_file_name))
+
+        # now check with an invalid signal
+        invalid_signal = 15 # invalid
+
+        with open(conv_signal_file_name, 'w') as conv_signal_file:
+            conv_signal_file.write(str(invalid_signal))
+
+        self.__CheckConvergenceSignalFile(invalid_signal) # to make sure that the file was written correctly
+
+        with self.assertRaisesRegex(RuntimeError, "Read an invalid convergence signal: {}, can only be 0 for non-convergence or 1 for convergence".format(invalid_signal)):
+            KratosCoSim.EMPIRE_API.EMPIRE_API_recvConvergenceSignal()
+
+        # manually deleting since the call above throws
+        os.remove(conv_signal_file_name)
+
+    def test_EMPIRE_API_sendMesh(self):
+        model = KM.Model()
+        model_part = model.CreateModelPart("For_Sending")
+
+        severity = KM.Logger.GetDefaultOutput().GetSeverity()
+        KM.Logger.GetDefaultOutput().SetSeverity(KM.Logger.Severity.WARNING) # mute MP-IO
+
+        model_part_io = KM.ModelPartIO(GetFilePath("generic_mdpa_files/Mok_CFD"))
+        model_part_io.ReadModelPart(model_part)
+
+        KM.Logger.GetDefaultOutput().SetSeverity(severity)
+
+        KratosCoSim.EMPIRE_API.EMPIRE_API_sendMesh(model_part)
+
+        # can directly use filecmp because there are no decimal-number issues
+        self.assertTrue(filecmp.cmp(GetFilePath("reference_files/EMPIRE_mesh_For_Sending.vtk_ref"), "EMPIRE_mesh_For_Sending.vtk"))
```

Comment:
comparison ... didn't you face any precision issues ?

Reply:
not in my specific case, but I think in the long run we still have to implement the comparison properly